### DISCUSSION TITLE
`.exclusive_system()` with `.after()`

### DISCUSSION THREAD

**Discussion by lukors** [bevyengine/bevy#3766] [Q&A] [answered]
If I use `.exclusive_system()` together with `.after()`, Bevy seems unable to find the label I put in `.after()`. Why is this?

The warning I get is something like `WARN bevy_ecs::schedule::graph_utils: my_project::my_file::my_system wants to be after unknown label: MyLabel`.

The code looks something like this:

```rust
fn build(&self, app: &mut App) {
    app.add_system_to_stage(
            CoreStage::Update,
            my_system.exclusive_system()
            .label(CustomStage::Apply)
            .after(CustomStage::MyLabel),
        );
}
```

Can exclusive systems not be after a certain label? Or do I need to do something different? Or is there something else going on?

**Comment by hymm** [comment, marked as answer]
This could be 1 of 2 problems
1. labels belong to a stage, so cannot be used across stages
2. labeling between exclusive and parallel systems is not allowed. If you want to order exclusive systems with parallel systems use the `at_start`, `before_commands`, `at_end` exclusive system modifiers. https://docs.rs/bevy/latest/bevy/ecs/system/struct.ExclusiveSystemFn.html#method.at_start

**Reply by lukors** [reply]
That's it, that labelling between exclusive and parallel systems is not allowed. I have no other exclusive system with that label, so that's why I'm getting the warning. Thanks a lot for the great answer! :smile:
